ext/standard/basic_functions.c

```diff
@@ -1856,7 +1856,7 @@ PHP_FUNCTION(getopt)
 
 	while ((o = php_getopt(argc, argv, opts, &php_optarg, &php_optind, 0, 1)) != -1) {
 		/* Skip unknown arguments. */
-		if (o == '?') {
+		if (o == -2) {
```

Comment:
Can you please add a `#define` for this magic value in main/php_getopt.h?

Reply:
Done.